doc/source/dev/missing-bits.rst

```diff
@@ -47,6 +47,15 @@ would probably do anyway even without the use of ``*``), *and* it means
 additional parameters can be added to the function anywhere after the
 ``*``; new parameters do not have to be added after the existing parameters.
 
+Return Objects
+~~~~~~~~~~~~~~
+For new functions or methods that return two or more conceptually distinct
+elements, use an object type that is not iterable. In particular, do not use
+``tuple``, ``namedtuple``, or ``make_tuple_bunch``, the latter of which is
+only for adding new attributes to the object returned by existing functions.
+This practice forces callers to be more explicit about the element of the
```

Comment:
```suggestion
only for adding new attributes to the object returned by existing functions.
Instead, use an existing return class (e.g. `OptimizeResult`), write a new
return class, or use `dataclasses.make_dataclass`.
This practice forces callers to be more explicit about the element of the
```
Will `dataclasses.make_dataclass` be recognized as an intersphinx link?
How should new custom dataclasses be documented?  That is, like this, where the name of the undocumented returned type is specified? (`BootstrapResult`, here.)
![image](https://user-images.githubusercontent.com/6570539/198854970-3bec9177-c6b4-409b-a8e3-1a25146fbc7f.png)

Reply:
> Will `dataclasses.make_dataclass` be recognized as an intersphinx link?

Yes, I think so.

> How should new custom dataclasses be documented? That is, like this, where the name of the undocumented returned type is specified?

That sounds good. Since the objects are not present in the public API, Sphinx won't link to them anyways. So, this seems like the best way to document them